storage/pebble/registers.go

```diff
@@ -35,9 +36,10 @@ func NewRegisters(db *pebble.DB) (*Registers, error) {
 
 	// All registers between firstHeight and lastHeight have been indexed
 	return &Registers{
-		db:           db,
-		firstHeight:  firstHeight,
-		latestHeight: atomic.NewUint64(latestHeight),
+		db:             db,
+		firstHeight:    atomic.NewUint64(firstHeight),
+		latestHeight:   atomic.NewUint64(latestHeight),
+		pruneThreshold: pruneThreshold,
```

Comment:
in your updates, you pass `0` for `pruneThreshold`, which I'm guessing is supposed to indicate that pruning is disabled. The logic so far in this file will treat that as "keep 0 blocks" and will prune everything.

I think we need logic in the constructor that checks if the value is 0 and updates it to maxint